### DISCUSSION TITLE
how can add a field to user table

### DISCUSSION THREAD

**Discussion by 0xlimon** [appwrite/appwrite#4996] [Q&A] [answered]
For auth I need to store more information for the user
How can I add a field to the user or auth table?

**Comment by stnguyen90** [comment, marked as answer]
You can use [preferences](https://appwrite.io/docs/client/account?sdk=web-default#accountUpdatePrefs) for non-sensitive data that the user can freely update. For data that other users should be able to fetch or the user shouldn't be able to edit, the typical approach is to create a custom collection for the data. 

There is an [issue](https://github.com/appwrite/appwrite/issues/4805) requesting this feature, though. Be sure to 👍🏼 it to upvote it.

**Reply by 0xlimon** [reply]
> You can use [preferences](https://appwrite.io/docs/client/account?sdk=web-default#accountUpdatePrefs) for non-sensitive data that the user can freely update. For data that other users should be able to fetch or the user shouldn't be able to edit, the typical approach is to create a custom collection for the data.
> 
> There is an [issue](https://github.com/appwrite/appwrite/issues/4805) requesting this feature, though. Be sure to 👍🏼 it to upvote it.

Is there no other way? I want there to be a field with a default value for the user without it being sent from their client side

**Reply by stnguyen90** [reply]
If you use a custom collection, you can create an [Appwrite Function](https://appwrite.io/docs/functions) that triggers on account creation to set the default value.

**Reply by polaroidkidd** [reply]
how do you go about creating a foreign key between a user and a custom collection in a custom database? Is there a method for it or do I have to get the user id in the appwrite function and then retrieve the entire user object through a join statement or in plain js?
